functions/src/api/user/index.ts

```diff
@@ -78,4 +83,21 @@ userRouter.put(
   },
 );
 
+userRouter.get('/:id', async ctx => {
+  try {
+    const userProfile = await getProfile(ctx.params.id);
```

Comment:
And maybe a new `getHost` that checks that the user actually has that claim before returning any data

Reply:
That was my initial approach, but then I remembered that there are private sessions 😬